main.go

```diff
@@ -0,0 +1,168 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"net/http"
+	"os"
+	awsproviderapi "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cloud-ingress-operator/pkg/cloudclient"
+	"github.com/openshift/cloud-ingress-operator/config"
+	"github.com/openshift/cloud-ingress-operator/pkg/ingresscontroller"
+	"github.com/openshift/cloud-ingress-operator/pkg/localmetrics"
+	baseutils "github.com/openshift/cloud-ingress-operator/pkg/utils"
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	osdmetrics "github.com/openshift/operator-custom-metrics/pkg/metrics"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	cloudingressmanagedopenshiftiov1alpha1 "github.com/openshift/cloud-ingress-operator/api/v1alpha1"
+	apischemecontroller "github.com/openshift/cloud-ingress-operator/controllers/apischeme"
+	publishingstrategycontroller "github.com/openshift/cloud-ingress-operator/controllers/publishingstrategy"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+var (
+	osdMetricsPort = "8181"
+	osdMetricsPath = "/metrics"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(cloudingressmanagedopenshiftiov1alpha1.AddToScheme(scheme))
+
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(machineapi.AddToScheme(scheme))
+	utilruntime.Must(ingresscontroller.AddToScheme(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+	utilruntime.Must(awsproviderapi.SchemeBuilder.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
```

Comment:
The old CIO performs leader-for-life election using `Become()`, we may need to consider if we are maintaining that behaviour.

Reply:
The leader election was added.